diff --git a/lib/src/tree_cursor.c b/lib/src/tree_cursor.c
--- a/lib/src/tree_cursor.c
+++ b/lib/src/tree_cursor.c
@@ -124,7 +124,7 @@ static inline bool ts_tree_cursor_child_iterator_previous(
 ) {
   // this is mostly a reverse `ts_tree_cursor_child_iterator_next` taking into
   // account unsigned underflow
-  if (!self->parent.ptr || (int8_t)self->child_index == -1) return false;
+  if (!self->parent.ptr || self->child_index == UINT32_MAX) return false;
   const Subtree *child = &ts_subtree_children(self->parent)[self->child_index];
   *result = (TreeCursorEntry) {
     .subtree = child,
